### ISSUE TITLE
Moq breaks when a dependency uses build tags to exclude duplicate code

### ISSUE THREAD

**Issue by YetAnotherMatt** [matryer/moq#47]
When running moq in a package that imports code using build tags, moq fails to compile if any imports include duplicate method declarations that would otherwise be resolved by build tags. 
For example `github.com/gorilla/mux/context_native.go` and `github.com/gorilla/mux/context_gorilla.go` both declare a `contextGet` method - one or other file should be excluded according to a build tag:
```
// +build !go1.7
```
Attempting to generate mocks against this file:
```
package moqtest

import "github.com/gorilla/mux"

//go:generate moq -out mock_test.go . MyInterface

type MyInterface interface {
	DoStuff() string
}

func foo() {

	router := mux.NewRouter()

	println(router)
}
```
results in the following error:
```
live.go:3:8: could not import github.com/gorilla/mux (/.../vendor/github.com/gorilla/mux/context_native.go:10:6: contextGet redeclared in this block)
moq [flags] destination interface [interface2 [interface3 [...]]]
  -out string
    	output file (default stdout)
  -pkg string
    	package name (default will infer)
```

**Comment by HaraldNordgren** [comment]
@YetAnotherMatt I think this one was solved by https://github.com/matryer/moq/pull/45

**Comment by matryer** [comment]
Assuming fixed by #45.
